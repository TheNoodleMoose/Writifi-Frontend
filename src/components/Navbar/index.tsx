import React from "react";
import { Link } from "react-router-dom";
import Button from "../Button";
import styled from "styled-components";
import { ReactComponent as LogoSvg } from "../../assets/svgs/Logo.svg";

const Navbar = () => {
  return (
    <NavContainer>
      <TitleLink to="/">
        <TitleContainer>
          <LogoSvg />
          <TitleText>ritifi</TitleText>
        </TitleContainer>
      </TitleLink>
      <Link to="/login">
        <Button text="Login" />
      </Link>
    </NavContainer>
  );
};

export default Navbar;

const NavContainer = styled.nav`
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin: 10px 0;
`;

const TitleContainer = styled.div`
  display: flex;
  align-items: flex-end;
  justify-content: flex-end;
`;

const TitleLink = styled(Link)`
  text-decoration: none;
`;

const TitleText = styled.p`
  font-size: 25px;
  color: #2c3e50;
  font-weight: bold;
  margin: 0;
  margin-left: -10px;
  text-decoration: none;
`;
