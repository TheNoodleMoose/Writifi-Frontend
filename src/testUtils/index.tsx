import React, { ReactElement, ReactNode } from "react";

import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

export const renderWithContext = (component: ReactElement) => {
  const history = createMemoryHistory();
  const wrapper = ({ children }: { children?: ReactNode }) => (
    <Router history={history}>{children}</Router>
  );

  const renderResults = render(component, { wrapper });
  return {
    ...renderResults,
    snapshot: () => {
      expect(renderResults.container).toMatchSnapshot();
    }
  };
};
