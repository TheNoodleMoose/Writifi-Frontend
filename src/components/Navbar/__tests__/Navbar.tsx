import React from "react";
import { renderWithContext } from "../../../testUtils";

import Navbar from "..";

it("renders correctly", () => {
  renderWithContext(<Navbar />).snapshot();
});
