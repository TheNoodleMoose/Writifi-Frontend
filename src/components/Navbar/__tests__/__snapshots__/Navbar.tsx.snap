// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav>
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/login"
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
